src/pages/User/UserDetailsEditForm.tsx

```diff
@@ -53,21 +54,28 @@ const formFields = (
 interface UserDetailsEditFormProps {
   userDetails: GetUserDetailsResponse;
   closeDialog: () => void;
+  mutation: UseMutationResult<UserDetails, AxiosError<unknown>, PutUserDetails, unknown>;
 }
 
-function UserDetailsEditForm({ userDetails, closeDialog }: UserDetailsEditFormProps) {
-  const { mutate, error, isLoading, isSuccess } = useUserDetailsEdit();
+function UserDetailsEditForm({
+  userDetails,
+  closeDialog,
+  mutation,
+}: UserDetailsEditFormProps) {
   const [errorMessage, setErrorMessage] = useState<string | null>(null);
   const [avatarImage, setAvatarImage] = useState<Blob | null>(null);
   const prepareInitialValues = useLoadImage(userDetails.avatarImage, setAvatarImage);
 
+  const { mutate, isLoading, isSuccess, error } = mutation;
+
   useEffect(() => {
     prepareInitialValues();
   }, [prepareInitialValues]);
 
   useEffect(() => {
     if (isSuccess) closeDialog();
-  }, [isSuccess, closeDialog]);
+    else setErrorMessage(getErrorMessage(error));
```

Comment:
może tak trochę precyzyjniej? czy celowo to tak zrobiłeś
```suggestion
    else if (error !== null) setErrorMessage(getErrorMessage(error));
```

Reply:
ok